### DISCUSSION TITLE
Make Shortcuts User Definable

### DISCUSSION THREAD

**Discussion by TheMakerTales** [hlorus/CAD_Sketcher#204] [Feature Requests]
Currently, it seems that CAD Sketcher has some hard-baked shortcuts that don't easily be changed. On top of that, some of them seem to override blender defaults. I know it's not a big deal as you don't use many blender defaults in the sketcher workspace but I see a future where this might be a problem.

I would love it if in the addon preferences we could define our own shortcuts a little like CAD-Transforms has done:
<img width="472" alt="Capture" src="https://user-images.githubusercontent.com/103122109/176142545-521a5e8d-db49-4c6c-98cd-525389639c4a.PNG">

Additionally, a little dropdown to pick from common CAD programs shortcut lists like fusion, freecad etc would be pretty interesting too.

**Comment by ginevike** [comment]
It would give a user flexibility however it would not be user friendly since nobody would be able to follow a tutorial or project steps due to the inconsistent key strokes. This is just my opinion, new guy on the block but many years working with other software that has very rigid structuring for key binds specifically for this reason.

**Comment by Intuos5** [comment]
Do note that almost no shortcuts are hardcoded, they can be changed under the CAD Sketcher header. That said, there's no way to batch edit multiple similar shortcuts at once (each tool has a separate entry listed). This has advantages too, so you can toggle between tools using the same shortcut, for instance. 
What I dislike about the CAD transforms editor is that it uses dropdowns throughout. I prefer to type in the hotkey in an input box instead. The reason being: often times, the addon maker does not include all possible hotkeys and that can be annoying if you have a custom hotkey scheme that you like to use.
